solutions/is_leap_year.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+A module for determining whether the given year is leap year or not
+
+Module contents:
+    - is_leap: determines whether the given year is leap year or not.
+
+Created on XX XX XX
+@author: Khusro Sakhi
+"""
+
+
+def is_leap_year(year) -> bool:
+    """Determines if the given year is a leap year.
+
+    Parameters:
+        year (int): The year to test.
+
+    Returns:
+        bool: True if the year is a leap year, False otherwise.
+
+     Raises:
+        AssertionError: If the input is a string.
+        AssertionError: If the input is a negative year.
+        AssertionError: If the input is None.
+        AssertionError: If the input is a float.
+
+    >>> is_leap(2000)
+    True
+
+    >>> is_leap(1990)
+    False
+
+    >>> is_leap(2100)
+    False
+
+    >>> is_leap(2004)
+    True
+    """
+    assert isinstance(year, int), "entered year is not an integer"
+    assert year >= 0, "year is less than 0"
```

Comment:
i dont know but is there a year zero? 
if not then this should be year > 0.

Reply:
Good catch, I searched about it and turns out year 0 is widely not accepted as a valid year 👍